### ISSUE TITLE
Possible to add bare repos?

### ISSUE THREAD

**Issue by HnryStGTkr** [nosarthur/gita#95]
Is it possible to add bare repos to gita?

For example, i have the following: 
```bash
alias dotgit="git --git-dir=$HOME/somefolder --work-tree=$HOME"
```
I'm unable to add the folder to gita though with `gita add <foldername>`

**Comment by nosarthur** [comment]
This is because I check `.git` folder/file to verify if the given folder is a git repo. 

I can add an option `-b` to add bare repo. Under the hood it can verify via 

```
git rev-parse --is-bare-repository 
```

**Comment by HnryStGTkr** [comment]
Oh, ok - that would be great thank you!

**Comment by nosarthur** [comment]
I forgot to mention that you can add the repo path directly to `~/.config/gita/repo_path` as a hack. 

I am not sure what git operations are not valid for bare repo. Let me know

**Comment by nosarthur** [comment]
It seems most git operations are not valid in a bare repo. What is the motivation to work on/view a bare repo?

**Comment by eriteric** [comment]
for an example use case: a lot of people put their dotfiles in a bare repo

**Comment by nosarthur** [comment]
What's the benefit of that, comparing to a regular repo?

**Comment by eriteric** [comment]
I probably don't know enough to qualify the legitimacy of bare repos to anyone.
I guess the benefit of a bare repo is to not have a working directory that takes up space. That's pretty much it.
https://www.atlassian.com/git/tutorials/dotfiles
There are a lot of people that do that with their dotfiles - so that should make it legitimate enough to consider it.
However, I didn't know this until recently but apparently the repo does not have to actually be bare as you can simply change the working directory of any git repository. As far as I know, you still have to tell git where the working directory is in each command though. like this: ` --git-dir=$HOME/.cfg/ --work-tree=$HOME` Is that something gita can already handle?

So, if the difficulty just lies with it being bare, that COULD be sidestepped if we provide people with information/instruction.

Another general use case for bare repos, and this is perhaps not within gita's domain-just explaining what they're used for, would be on a live server where a bare repo is used with a post-receive hook to send the files elsewhere in the filesystem.

**Comment by nosarthur** [comment]
Thanks @eriteric . It's an interesting trick

If you want to add extra parameters, you can try `gita super` or `gita shell`. For example,

```
gita super bare-repo status --git-dir=$HOME/.cfg/ --work-tree=$HOME
```

As a hack to add bare repo, you can edit the `~/.config/gita/repo_path` directly. It's in a simple csv format

```
path,name
```

https://www.geeksforgeeks.org/bare-repositories-in-git/
> The only possible operations on the Bare Repository are Pushing or Cloning.

**Comment by eriteric** [comment]
I see, thanks for the link. It outlines another reason for bare repos: Pushing to a non-bare repo would make its working tree inconsistent. Makes sense...kind of lol.

Ok, btw I tried manually adding a bare repo to ~/.config/gita/repo_path and it does not show up in `gita ll`.

Putting the bare repo stuff aside, would you have any interest in it if I made a new issue for making it so that the flags don't have to be entered every command for **normal** repos that have special working directories? (perhaps making those things assignable in the config).

**Comment by nosarthur** [comment]
> Ok, btw I tried manually adding a bare repo to ~/.config/gita/repo_path and it does not show up in gita ll.

Without a worktree, the `git status` doesn't work either ...
```
10:41 dzhou (BARE:master) bare-repo $ g status
fatal: this operation must be run in a work tree
```

Custom git flags sounds like a good idea. Currently, the repo config file `repos.csv` has 3 fields: path, name, type. We can add a new field for flags.

**Comment by eriteric** [comment]
Alright great, I've added that. #151

**Comment by nosarthur** [comment]
After #151 is in, it will be easy to support bare repo
